### PR TITLE
[MRG] Addressed comments remained for check proper cut_coords in plot_stat_map

### LINKED ISSUE #784: check for proper coords on plot_stat_map
currently plot_stat_map will crash if a scalar is passed to cut_coords in single-axis display mode:

```
Traceback (most recent call last):
  File "mk_tsne_figure.py", line 57, in <module>
    draw_cross=False,display_mode='z',colorbar=False,cut_coords=0.)
  File "/Users/poldrack/anaconda/lib/python2.7/site-packages/nilearn-0.1.5.dev0-py2.7.egg/nilearn/plotting/img_plotting.py", line 891, in plot_stat_map
    **kwargs)
  File "/Users/poldrack/anaconda/lib/python2.7/site-packages/nilearn-0.1.5.dev0-py2.7.egg/nilearn/plotting/img_plotting.py", line 163, in _plot_img_with_bg
    colorbar=colorbar)
  File "/Users/poldrack/anaconda/lib/python2.7/site-packages/nilearn-0.1.5.dev0-py2.7.egg/nilearn/plotting/displays.py", line 400, in init_with_figure
    cut_coords = cls.find_cut_coords(img, threshold, cut_coords)
  File "/Users/poldrack/anaconda/lib/python2.7/site-packages/nilearn-0.1.5.dev0-py2.7.egg/nilearn/plotting/displays.py", line 950, in find_cut_coords
    n_cuts=cut_coords)
  File "/Users/poldrack/anaconda/lib/python2.7/site-packages/nilearn-0.1.5.dev0-py2.7.egg/nilearn/plotting/find_cuts.py", line 213, in find_cut_slices
    spacing = max(int(.5 / n_cuts * data.shape[axis]), 1)
ZeroDivisionError: float division by zero
```
should check to see if the arguments to cut_coords are a list, and coerce single coord into a list if passed as a scalar.

### PR DESCRIPTION
**Pull request by KamalakerDadi**
Should fix #784

### CODE REVIEW

File: nilearn/plotting/find_cuts.py [hunk 1]
```diff
@@ -196,6 +196,10 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
     orig_data = np.abs(img.get_data())
     this_shape = orig_data.shape[axis]
 
+    if not isinstance(n_cuts, numbers.Number):
+        raise ValueError("The number of cuts should be an integer. "
```
Comment (lesteve) [inline nilearn/plotting/find_cuts.py]: a strictly positive integer

File: nilearn/plotting/find_cuts.py [hunk 2]
```diff
@@ -214,15 +218,13 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
     epsilon = np.finfo(np.float32).eps
     difference = abs(round(n_cuts) - n_cuts)
-    message = ("The number of cuts in the given direction %s must be "
-               "an integer which should be greater than 0 and "
-               "less than or equal to %d. You provided n_cuts=%s " % (
-                   direction, this_shape, n_cuts))
-    if not isinstance(n_cuts, numbers.Number) or n_cuts < epsilon:
+    if round(n_cuts) < 0.5 or difference > epsilon:
+        message = ("The number of cuts in the given direction %s must be "
+                   "an integer which should be greater than 0 and "
```
Comment (lesteve) [inline nilearn/plotting/find_cuts.py]: Use `round(n_cuts) < 1.` for clarity

**Comment by bthirion** [comment]
Lgtm overall.

File: nilearn/plotting/find_cuts.py [hunk 3]
```diff
@@ -196,6 +196,10 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
     orig_data = np.abs(img.get_data())
```
Comment (bthirion) [inline nilearn/plotting/find_cuts.py]: Not related to this pr, but for n_cuts I would take a default value of,say, 5. 12 is huge.

Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: > but for n_cuts I would take a default value of,say, 5. 12 is huge.

+1. And I don't mind breaking backward compat at this stage of the
package.

Comment (KamalakerDadi) [inline nilearn/plotting/find_cuts.py]: Ok. So, I change the default to 5

Comment (GaelVaroquaux) [inline nilearn/plotting/find_cuts.py]: > So, I change the default to 5

Make sure you add this in doc/whats_new.rst

File: nilearn/plotting/find_cuts.py [hunk 4]
```diff
@@ -214,15 +218,13 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
...
     epsilon = np.finfo(np.float32).eps
     difference = abs(round(n_cuts) - n_cuts)
-    message = ("The number of cuts in the given direction %s must be "
-               "an integer which should be greater than 0 and "
-               "less than or equal to %d. You provided n_cuts=%s " % (
-                   direction, this_shape, n_cuts))
-    if not isinstance(n_cuts, numbers.Number) or n_cuts < epsilon:
+    if round(n_cuts) < 1. or difference > epsilon:
+        message = ("The number of cuts in the given direction %s must be "
+                   "an integer which should be greater than 0 and "
```
Comment (lesteve) [inline nilearn/plotting/find_cuts.py]: although equivalent for integers maybe "greater than or equal to 1" is more clear than "greater than 0"

Comment (AlexandreAbraham) [inline nilearn/plotting/find_cuts.py]: I would go for a more straightforward message.
`'Image has %d slices in direction %s. Therefore the number of cuts must be between 1 and %d. You provided n_cuts=%d' % (this_shape, direction, this_shape, n_cuts)`

File: nilearn/plotting/find_cuts.py [hunk 5]
```diff
@@ -196,6 +196,10 @@ def find_cut_slices(img, direction='z', n_cuts=12, spacing='auto'):
     orig_data = np.abs(img.get_data())
     this_shape = orig_data.shape[axis]
 
+    if not isinstance(n_cuts, numbers.Number):
+        raise ValueError("The number of cuts should be a strictly positive integer. "
+                         "You provided n_cuts=%s " % n_cuts)
```
Comment (AlexandreAbraham) [inline nilearn/plotting/find_cuts.py]: The error message does not match the condition: Here you are interested in the type. "should" is not strong enough, this is a hard constraints. Maybe something like:
`The number of cuts (n_cuts) must be an integer greater or equal than 1. You provided a value of type %s.`

File: nilearn/plotting/find_cuts.py (hunk 4 above)
Comment (lesteve) [inline nilearn/plotting/find_cuts.py]: Sounds good to me.

File: nilearn/plotting/find_cuts.py (hunk 5 above)
Comment (lesteve) [inline nilearn/plotting/find_cuts.py]: :+1:

File: nilearn/plotting/find_cuts.py (hunk 3 above)
Comment (lesteve) [inline nilearn/plotting/find_cuts.py]: Maybe use this opportunity to set its default to the same value than in displays (looks like it's 7) from https://github.com/nilearn/nilearn/blob/de38fe03a535b79b4250794833fe4049730e7ce7/nilearn/plotting/displays.py#L939.

File: nilearn/plotting/find_cuts.py (hunk 5 above)
Comment (lesteve) [inline nilearn/plotting/find_cuts.py]: Tiny comment it looks like the fully correct version is "greater **than** or equal **to** 1" and the alternative winning the google battle is "greater or equal **to** 1". See http://english.stackexchange.com/questions/38287/do-we-say-is-greater-or-equal-to-or-is-greater-or-equal-than for more details.

**Comment by lesteve** [comment]
Merging thanks..

**Pull request merged**